src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -14,13 +15,13 @@ namespace System.Runtime.CompilerServices
     /// </summary>
     /// <typeparam name="T">The type of the collection element.</typeparam>
     [Serializable]
+    [System.Diagnostics.DebuggerDisplay("Count = {Count}")]
```

Comment:
I'm not sure if this counts as an API change, but it does change the developer experience of using the class.
That said, I do think it's a very good idea.